packages/core/src/request-builder/header-builder/auth-headers.ts

```diff
@@ -0,0 +1,162 @@
+    );
+  }
+
+  const usableTokens = authTokens.filter(
+    (token: DestinationAuthToken) => !token.error
+  );
+
+  if (!usableTokens.length) {
+    throw Error(
+      [
+        'The destination tried to provide authorization tokens but failed in all cases. This is most likely due to misconfiguration.',
+        'Original error messages:',
+        ...authTokens.map(token => token.error)
+      ].join('\n')
+    );
+  }
+  const authToken = usableTokens[0];
+  return `${authToken.type} ${authToken.value}`;
+}
+
+async function headerFromOAuth2ClientCredentialsDestination(
+  destination: Destination
+): Promise<string> {
+  const response = await getOAuth2ClientCredentialsToken(destination).catch(
+    error => {
+      throw errorWithCause(
+        'Request for "OAuth2ClientCredentials" authentication access token failed or denied.',
+        error
+      );
+    }
+  );
+  return `Bearer ${response.access_token}`;
+}
+
+function headerFromBasicAuthDestination(destination: Destination): string {
+  if (isNullish(destination.username) || isNullish(destination.password)) {
+    throw Error(
+      'AuthenticationType is "BasicAuthentication", but "username" and / or "password" are missing!'
+    );
+  }
+
+  return basicHeader(destination.username, destination.password);
+}
+
+export function basicHeader(username: string, password: string): string {
+  return `Basic ${Buffer.from(`${username}:${password}`).toString('base64')}`;
+}
+
+function headerForPrincipalPropagation(destination: Destination): MapType<any> {
+  const principalPropagationHeader =
+    destination?.proxyConfiguration?.headers?.[
+      'SAP-Connectivity-Authentication'
+    ];
+  if (!principalPropagationHeader) {
+    throw Error(
+      'Principal propagation was selected in destination, but no SAP-Connectivity-Authentication bearer header was added by connectivity service.'
+    );
+  }
+  return {
+    'SAP-Connectivity-Authentication': principalPropagationHeader
+  };
+}
+
+function headerForProxy(destination: Destination): MapType<any> {
+  const proxyAuthHeader =
+    destination?.proxyConfiguration?.headers?.['Proxy-Authorization'];
+
+  return proxyAuthHeader ? { 'Proxy-Authorization': proxyAuthHeader } : {};
+}
+
+function legacyNoAuthOnPremiseProxy(destination: Destination): MapType<any> {
+  logger.warn(
+    `You are using \'NoAuthentication\' in destination: ${destination.name} which is an OnPremise destination. This is a deprecated configuration, most likely you wanted to set-up \'PrincipalPropagation\' so please change the destination property to the desired authentication scheme.`
+  );
+
+  let principalPropagationHeader;
+  try {
+    principalPropagationHeader = headerForPrincipalPropagation(destination);
+  } catch (e) {
+    logger.warn('No principal propagation header found.');
+  }
+
+  // TODO the proxy header are for OnPrem auth and are now handled correctly in the authorization-header.ts and should be removed here
```

Comment:
Could you move this TODO to the `getProxyRelatedAuthHeaders` method and say that the whole `legacyNoAutOnPremiseProxy`  method should be removed in version 2.0.